x-pack/metricbeat/module/azure/azure.go

```diff
@@ -87,6 +88,18 @@ func NewMetricSet(base mb.BaseMetricSet) (*MetricSet, error) {
 // It publishes the event which is then forwarded to the output. In case
 // of an error set the Error field of mb.Event or simply call report.Error().
 func (m *MetricSet) Fetch(report mb.ReporterV2) error {
+	// Set the reference time for the current fetch.
+	//
+	// The reference time is used to calculate time intervals
+	// and compare with collection info in the metric
+	// registry to decide whether to collect metrics or not,
+	// depending on metric time grain (check `MetricRegistry`
+	// for more information).
+	//
+	// We truncate the reference time to the second to avoid
```

Comment:
nit: i think it's worth being specific about what the millisecond issue is. e.g. 

```
We truncate the reference time to the second to avoid millisecond
variations in the collection period causing skipped collections. 
```

Reply:
Sweet, adding it!